### ISSUE TITLE
[Feature Request] slim down dev_setup script

### ISSUE THREAD

**Issue by sblackshear** [diem/move#25]
The dev_setup script installs many things (terraform, kubernetes, awscli, ...) that aren't needed to run Move. I think we should just need Rust + the .net and z3 deps for the prover.

**Comment by 666lcz** [comment]
As mentioned in the tutorial,

https://github.com/diem/move/blob/d5cb0578e5a8dc8afe2e6efb7504bb9ec1e63651/language/documentation/tutorial/README.md?plain=1#L34-L39


terraform, kubernetes, awscli will not get installed if running with `-ypt` because of 

https://github.com/diem/move/blob/d5cb0578e5a8dc8afe2e6efb7504bb9ec1e63651/scripts/dev_setup.sh#L765-L775

https://github.com/diem/move/blob/d5cb0578e5a8dc8afe2e6efb7504bb9ec1e63651/scripts/dev_setup.sh#L920-L940

@sblackshear do we still want to remove these code given that these flags are configurable?

**Comment by sblackshear** [comment]
Yes, I think we should remove both the logic and dev setup and the flags--better to have unused functionality deleted altogether rather than gated via flags IMO.

**Comment by 666lcz** [comment]
@sblackshear : https://github.com/diem/move/pull/73 is ready for review
